src/control/lib/netdetect/netdetect.go

```diff
@@ -507,10 +511,20 @@ func GetAffinityForNetworkDevices(deviceNames []string) ([]DeviceAffinity, error
 	return affinity, nil
 }
 
-// GetDeviceAlias is a complete method to find an alias for the device name provided.
-// For example, the device alias for "ib0" is a sibling node in the hwloc topology
-// with the name "hfi1_0".
+// GetDeviceAlias is a wrapper for getDeviceAliasWithSystemList.  This interface
+// specifies an empty additionalSystemDevices list which allows for the default behavior we want
+// for normal use.  Test functions can call getDeviceAliasWithSystemList() directly and specify
+// an arbitrary additionalSystemDevice list as necessary.
 func GetDeviceAlias(device string) (string, error) {
+	additionalSystemDevices := []string{}
```

Comment:
So is the functionality to call the real hardware to fetch devices to be added in a later patch?

Reply:
The functionality fetch the real hardware devices is built into the init function that is called in all cases.  The 'additionalSystemDevices' are added to that existing list of devices.  If you don't add any, you get only the real devices found.  If you do add devices, which is only done for testing, you get "known hardware devices and your additional devices".

Reply:
Ah, okay. I guess I didn't follow that the new parameter was only added for test purposes.

With that knowledge, I have a suggestion - How about having GetDeviceAlias run initDeviceScan() and pass the deviceConfig into the underlying helper? In that case, the deviceConfig could be already populated with the list of desired devices, so there would be no need for an extra param or extra logic to do the testing. Then GetDeviceAlias would be responsible for doing the real hardware stuff, and the underlying helper would just process the incoming information. I think in that case the underlying function name should change (getDeviceAliasFromConfig, perhaps?).

Wouldn't block on the change but it may clarify the code a bit.

Reply:
The problem with having GetDeviceAlias run initDeviceScan() is that it would require the test code that calls getDeviceAliasWithSystemList() to also run initDeviceScan() which duplicates code in the test functions.  I modified GetDeviceAlias to eliminate this code duplication.

Reply:
I guess I don't totally understand why the test code would need to run initDeviceScan(). Couldn't the test just set up a DeviceScan object in such a way as to exercise the different code paths in getDeviceAliasWithSystemList()? Or is the setup more complex than it looks?

In either case I think your current changeset is a good way to make the logic more testable, no need to churn on it if it's impractical to go further right now.